Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/README.md

```diff
@@ -0,0 +1,907 @@
+but it does require preparation.
+
+This guide aims to help you automated acquisition of models from Hugging Face, minimize time spent optimizing models for
+TensorRT, and configuring automatic scaling and load balancing for your models. This guide does not cover Kubernetes'
+basics, secure ingress/egress from your cluster to external clients, nor cloud provider interfaces or implementations of
+Kubernetes.
+
+We'll cover the following topics:
+
+* [Cluster Setup](#cluster-setup)
+  * [Core Cluster Services](#core-cluster-services)
+    * [Kubernetes Node Feature Discovery service](#kubernetes-node-feature-discovery-service)
+    * [NVIDIA Device Plugin for Kubernetes](#nvidia-device-plugin-for-kubernetes)
+    * [NVIDIA GPU Feature Discovery service](#nvidia-gpu-feature-discovery-service)
+  * [Metrics Collection Services](#metrics-collection-services)
+    * [Create a Monitoring Namespace](#create-a-monitoring-namespace)
+    * [Prometheus Services](#prometheus-services)
+    * [NVIDIA Data Center GPU Manager (DCGM) Exporter](#nvidia-data-center-gpu-manager-dcgm-exporter)
+    * [Connect DCGM and Triton Metrics to Prometheus](#connect-dcgm-and-triton-metrics-to-prometheus)
+    * [Triton Metrics Prometheus Rule](#triton-metrics-prometheus-rule)
+  * [Hugging Face Authorization](#hugging-face-authorization)
+* [Triton Preparation](#triton-preparation)
+  * [Model Preparation Script](#model-preparation-script)
+  * [Custom Container Image](#custom-container-image)
+  * [Kubernetes Pull Secrets](#kubernetes-pull-secrets)
+* [Triton Deployment](#triton-deployment)
+  * [Deploying Single GPU Models](#deploying-single-gpu-models)
+  * [Deploying Models Too Large for a Single GPU](#deploying-models-too-large-for-a-single-gpu)
+  * [Utilizing Multiple GPU SKUs](#utilizing-multiple-gpu-skus)
+  * [Monitoring Triton in Kubernetes](#monitoring-triton-in-kubernetes)
+* [Developing this Guide](#developing-this-guide)
+
+Prior to beginning this guide/tutorial you will need a couple of things.
+
+* Kubernetes Control CLI (`kubectl`)
+  [ [documentation](https://kubernetes.io/docs/reference/kubectl/introduction/)
+  | [download](https://kubernetes.io/releases/download/) ]
+* Helm CLI (`helm`)
+  [ [documentation](https://helm.sh/)
+  | [download](https://helm.sh/docs/intro/install) ]
+* Docker CLI (`docker`)
+  [ [documentation](https://docs.docker.com/)
+  | [download](https://docs.docker.com/get-docker/) ]
+* Decent text editing software for editing YAML files.
+* Kubernetes cluster.
+* Fully configured `kubectl` with administrator permissions to the cluster.
+
+## Cluster Setup
+
+The following instructions are setting up Horizontal Pod Autoscaling (HPA) for Triton Server in a Kubernetes cluster.
+
+
+### Prerequisites
+
+This guide assumes that all nodes with NVIDIA GPUs have the following:
+- A node label of `nvidia.com/gpu=present` to more easily identify nodes with NVIDIA GPUs.
```

Comment:
the point wasn to not soley rely on the gpu-feature-discovery service. thus allow us control over where the NVIDIA services interact because a few of them fail when no GPU is present.

Reply:
It does seem like we'd want to proscribe an automated  way to handle this - is there a way to use either the label added by gpu feature discovery - or the manual added label? - can be for the future.